src/modules/auth/auth.service.ts

```diff
@@ -29,7 +29,7 @@ export class AuthService {
     return user.toObject() as User;
   }
 
-  async login(loginDto: LoginDto): Promise<User> {
+  async login(loginDto: LoginDto): Promise<{ user: User; token: string }> {
     const foundUser = await this.userModel.findOne({ email: loginDto.email });
     if (!foundUser) {
       throw new UnauthorizedException('User not found');
```

Comment:
We shouldn't tell user User not found. Change this error message to a more general one. Then return 401 code not 500